src/libcadet/model/parts/ConvectionDispersionOperator.cpp

```diff
@@ -185,29 +186,21 @@ bool ConvectionDispersionOperatorBase::configure(UnitOpIdx unitOpIdx, IParameter
  */
 bool ConvectionDispersionOperatorBase::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
-	double prevVelocity = static_cast<double>(_curVelocity);
-
 	// If we don't have cross section area, velocity is given by parameter
+	double _dir_old = _dir;
 	if (_crossSection <= 0.0)
-		_curVelocity = getSectionDependentScalar(_velocity, secIdx);
+		_dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
 	else if (!_velocity.empty())
 	{
-		if (secIdx > 0)
-		{
-			const double dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx - 1));
-			if (dir < 0.0)
-				prevVelocity *= -1.0;
-		}
-
 		// We have both cross section area and interstitial flow rate
 		// _curVelocity has already been set to the network flow rate in setFlowRates()
 		// the direction of the flow (i.e., sign of _curVelocity) is given by _velocity
-		const double dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
-		if (dir < 0.0)
+		_dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		if (_dir_old * static_cast<double>(_dir) < 0.0)
```

Comment:
The `static_cast` is not necessary as `_dir` is already of type `double`.

Reply:
Actually, if I don't cast, I get the following error:
```
  193 |   double dir_new = getSectionDependentScalar(_velocity, secIdx);
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
      |                                             |
      |                                             const sfad::Fwd<double>
/home/jo/software/cadet/CADET/src/libcadet/model/parts/ConvectionDispersionOperator.cpp:201:45: error: cannot convert ‘const sfad::Fwd<double>’ to ‘double’ in initialization
  201 |   double dir_new = getSectionDependentScalar(_velocity, secIdx);
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
      |                                             |
      |                                             const sfad::Fwd<double>
```

Reply:
Sorry for the confusion:
* The cast **is required** here: `_dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));`
* It is **not necessary** here: `if (_dir_old * static_cast<double>(_dir) < 0.0)`